src/stan/mcmc/chains.hpp

```diff
@@ -292,6 +291,91 @@ namespace stan {
         return ess;
       }
 
+      /**
+       * Returns the effective sample size for the specified parameter
+       * across all kept samples.
+       *
+       * The implementation is close to the effective sample size
+       * description in BDA3 (p. 286-287).  See more details in Stan
+       * reference manual section "Effective Sample Size".
+       *
+       * Current implementation takes the minimum number of samples
+       * across chains as the number of samples per chain.
+       *
+       * @param std::vector<std::vector<double> > samples
+       * @return
+       */
+      double
+      effective_sample_size(const std::vector<std::vector<double> >& samples)
```

Comment:
We have to make sure with @bgoodri and @ariddell that this will work for both RStan and PyStan.  I was proposing a POD solution with just this signature:

```
double effective_sample_size(const double* draws, int n_chains, int n_draws_per_chain);  
```

I don't think we can use 2D arrays because we don't know sizes statically.

where the chains were stored one after the other.

Reply:
Right, a POD solution.  I backed away from that after thinking that the [original issue](https://github.com/stan-dev/stan/issues/2462) had some agreement that the std::vector signature would suffice and finding so many stan::math functions based on std::vectors.  But, you're right, confirmation would be good.  I'll wait until we hear from @bgoodri and/or @ariddell.  Happy to rewrite as you all prefer.